core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -122,7 +122,10 @@ class NewHadoopRDD[K, V](
       case _ =>
     }
     val jobContext = new JobContextImpl(_conf, jobId)
-    val rawSplits = inputFormat.getSplits(jobContext).toArray
+    var rawSplits = inputFormat.getSplits(jobContext).toArray(Array.empty[InputSplit])
+    if (sparkContext.getConf.getBoolean("spark.hadoop.filterOutEmptySplit", false)) {
+      rawSplits = rawSplits.filter(_.getLength>0)
```

Comment:
Space around operator.
You should filter before making an array.

Reply:
Is there any one use empty file to do something ?
for example:
sc.textFile("/somepath/*").mapPartitions(....)
setting this flag to true by default may change the behavior of user's application.